src/hotspot/share/opto/vectornode.hpp

```diff
@@ -1607,11 +1608,23 @@ class VectorRearrangeNode : public VectorNode {
     // assert(mask->is_VectorMask(), "VectorBlendNode requires that third argument be a mask");
   }
 
+  Node* Ideal(PhaseGVN* phase, bool can_reshape);
   virtual int Opcode() const;
   Node* vec1() const { return in(1); }
   Node* vec_shuffle() const { return in(2); }
 };
 
+
+class SelectFromTwoVectorNode : public VectorNode {
+public:
+  SelectFromTwoVectorNode(Node* in1, Node* in2, Node* in3, const TypeVect* vt)
+  : VectorNode(in1, in2, in3, vt) {}
```

Comment:
I would prefer more expressive variable names and a short specification what the node does. Otherwise one always has to reverse-engineer what inputs are acceptable etc. I mean you could even require `VectorNode*` as inputs.